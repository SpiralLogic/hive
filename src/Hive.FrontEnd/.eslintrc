{
  "plugins": [
    "@typescript-eslint",
    "react-hooks",
    "testing-library",
    "jest-dom",
    "unicorn"
  ],
  "env": {
    "browser": true,
    "es6": true
  },
  "extends": [
    "kentcdodds",
    "plugin:jest/recommended",
    "plugin:jest-dom/recommended",
    "plugin:prettier/recommended",
    "plugin:testing-library/react",
    "plugin:testing-library/dom",
    "prettier",
    "plugin:unicorn/recommended",
    "preact"
  ],
  "rules": {
    "unicorn/no-null": "off",
    "no-unused-vars": "off",
    "@typescript-eslint/no-unused-vars": [
      "error"
    ]
  },
  "globals": {
    "Atomics": "readonly",
    "SharedArrayBuffer": "readonly",
    "h": true
  },
  "parser": "@typescript-eslint/parser",
  "parserOptions": {
    "project": "./tsconfig.json",
    "jsx": true
  },
  "overrides": [
    {
      "files": [
        "**/*.jsx",
        "**/*.tsx"
      ],
      "rules": {
        "unicorn/filename-case": "off"
      }
    }
  ],
  "settings": {
    "react": {
      "pragma": "h"
    },
    "propWrapperFunctions": [
      "forbidExtraProps",
      {
        "property": "freeze",
        "object": "Object"
      }
    ],
    "linkComponents": [
      "Hyperlink",
      {
        "name": "Link",
        "linkAttribute": "to"
      }
    ]
  },
  "ignorePatterns": [
    "/node_modules/*/**",
    "/*.config.js",
    "/src/public/*/**",
    "**/__coverage__/**",
    "/src/index.tsx",
    "/src/index.html",
    "/build.js",
    "/inject.shim.js"
  ]
}
