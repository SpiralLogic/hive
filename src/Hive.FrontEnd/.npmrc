node-version=20.11.1