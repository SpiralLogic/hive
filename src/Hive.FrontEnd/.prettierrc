{
  "singleQuote": true,
  "printWidth": 110,
  "proseWrap": "always",
  "bracketSameLine": true,
  "endOfLine": "auto",
  "trailingComma": "es5",
  "tabWidth": 2
}
