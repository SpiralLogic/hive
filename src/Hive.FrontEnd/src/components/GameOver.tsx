import '../css/share.css';

import { FunctionComponent } from 'preact';

const GameOver: FunctionComponent<{ outcome: string }> = ({ outcome }) => (
  <>
    <p>{outcome}</p>
    <button autofocus title="New Game" onClick={() => window.location.assign(`/`)}>
      New Game
    </button>
  </>
);
GameOver.displayName = 'GameOver';
export default GameOver;
