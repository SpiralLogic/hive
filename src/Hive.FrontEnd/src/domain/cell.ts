import { HexCoordinates } from './hex-coordinates';
import { Tile } from './tile';

export type Cell = {
  readonly coords: HexCoordinates;
  tiles: Array<Tile>;
};
