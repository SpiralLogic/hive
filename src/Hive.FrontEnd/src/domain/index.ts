export * from './cell';
export * from './game-state';
export * from './hex-coordinate';
export * from './move';
export * from './player';
export * from './row';
export * from './tile';
export * from './creature';
