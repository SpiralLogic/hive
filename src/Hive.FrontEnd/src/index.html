<!DOCTYPE html>
<html lang="en">
<head>
  <meta charset="UTF-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta name="description" content="Hive online board game">
  <title>Hive</title>
  [css]
  [scripts]    
</head>
</html>
