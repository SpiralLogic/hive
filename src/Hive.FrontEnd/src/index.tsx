import { h, render } from 'preact';
import App from './components/App';
import { serverConnectionFactory } from './services';
import creatures from './svg/creatures.svg?raw';

import GameEngine from './services/game-engine';

document.body.insertAdjacentHTML('beforeend', `${creatures}`);
document.title = import.meta.env.HIVE_APP_TITLE;

const [, , gameId, currentPlayer] = window.location.pathname.split('/');
const engine = new GameEngine({ gameId, currentPlayer });
const properties = {
  engine,
  connectionFactory: serverConnectionFactory,
};

render(h(App, properties), document.body);

export { removeOtherPlayerMoves } from './utilities/hextille';
export { cellKey } from './utilities/hextille';
console.log(import.meta.env);
