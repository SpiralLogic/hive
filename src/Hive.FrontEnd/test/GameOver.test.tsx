import { render, screen } from '@testing-library/preact';
import userEvent from '@testing-library/user-event';
import GameOver from '../src/components/GameOver';
import { GameStatus } from '../src/domain';
import { mockLocation, renderElement } from './test-helpers';

describe('<GameOver>', () => {
  test('close calls base url', () => {
    const location = { assign: jest.fn() };
    const restore = mockLocation(location);
    render(<GameOver outcome="Draw" />);

    userEvent.click(screen.getByRole('button'));
    expect(location.assign).toBeCalledWith('/');
    restore();
  });

  const outcomes: GameStatus[] = [
    'NewGame',
    'MoveSuccess',
    'AiWin',
    'Player0Win',
    'Player1Win',
    'MoveSuccessNextPlayerSkipped',
    'MoveInvalid',
    'GameOver',
    'Draw',
  ];

  for (const outcome of outcomes) {
    it(`${outcome} snapshot player 0`, () => {
      expect(renderElement(<GameOver outcome={outcome} />)).toMatchSnapshot();
    });

    it(`${outcome} snapshot player 1`, () => {
      expect(renderElement(<GameOver outcome={outcome} />)).toMatchSnapshot();
    });
  }
});
