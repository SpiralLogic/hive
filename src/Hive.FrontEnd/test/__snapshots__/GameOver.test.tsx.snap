// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gameOver snapshot tests AiWin snapshot player 0 1`] = `
<body>
  <div>
    <p>
      AiWin
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests AiWin snapshot player 1 1`] = `
<body>
  <div>
    <p>
      AiWin
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests Draw snapshot player 0 1`] = `
<body>
  <div>
    <p>
      Draw
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests Draw snapshot player 1 1`] = `
<body>
  <div>
    <p>
      Draw
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests GameOver snapshot player 0 1`] = `
<body>
  <div>
    <p>
      GameOver
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests GameOver snapshot player 1 1`] = `
<body>
  <div>
    <p>
      GameOver
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests MoveInvalid snapshot player 0 1`] = `
<body>
  <div>
    <p>
      MoveInvalid
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests MoveInvalid snapshot player 1 1`] = `
<body>
  <div>
    <p>
      MoveInvalid
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests MoveSuccess snapshot player 0 1`] = `
<body>
  <div>
    <p>
      MoveSuccess
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests MoveSuccess snapshot player 1 1`] = `
<body>
  <div>
    <p>
      MoveSuccess
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests MoveSuccessNextPlayerSkipped snapshot player 0 1`] = `
<body>
  <div>
    <p>
      MoveSuccessNextPlayerSkipped
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests MoveSuccessNextPlayerSkipped snapshot player 1 1`] = `
<body>
  <div>
    <p>
      MoveSuccessNextPlayerSkipped
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests NewGame snapshot player 0 1`] = `
<body>
  <div>
    <p>
      NewGame
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests NewGame snapshot player 1 1`] = `
<body>
  <div>
    <p>
      NewGame
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests Player0Win snapshot player 0 1`] = `
<body>
  <div>
    <p>
      Player0Win
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests Player0Win snapshot player 1 1`] = `
<body>
  <div>
    <p>
      Player0Win
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests Player1Win snapshot player 0 1`] = `
<body>
  <div>
    <p>
      Player1Win
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;

exports[`gameOver snapshot tests Player1Win snapshot player 1 1`] = `
<body>
  <div>
    <p>
      Player1Win
    </p>
    <button
      autofocus=""
      title="New Game"
    >
      Close
    </button>
  </div>
</body>
`;
